client/index.js

```diff
@@ -9,12 +9,22 @@ import { render } from '@wordpress/element';
  */
 import './stylesheets/_index.scss';
 import { PageLayout, EmbedLayout, PrimaryLayout as NoticeArea } from './layout';
+import Navigation from './navigation';
 import 'wc-api/wp-data-store';
 import { withSettingsHydration } from '@woocommerce/data';
 
 const appRoot = document.getElementById( 'root' );
+const navigationRoot = document.getElementById( 'woocommerce-embedded-navigation' );
 const settingsGroup = 'wc_admin';
 
+if ( navigationRoot ) {
+	render( <Navigation />, navigationRoot );
```

Comment:
Lets use the new Settings store so we don't have to use `import { getSetting } from '@woocommerce/wc-admin-settings';`. See below `withSettingsHydration` so that the navigation component can always have the most up to date settings. You can retrieve the settings with `getSettingsForGroup` selector.

If something dynamically alters the `wcNavigation`, we can receive updates without a page refresh.

Reply:
Awesome- updated in https://github.com/woocommerce/woocommerce-admin/pull/3202/commits/c896aab78b371cc82459845c23c2892ad58de372